subsys/net/lib/ftp_client/src/ftp_client.c

```diff
@@ -0,0 +1,735 @@
+		ret = -errno;
+	}
+
+	if (client.sec_tag > 0) {
+		sec_tag_t sec_tag_list[1] = { client.sec_tag };
+
+		ret = setsockopt(data_sock, SOL_TLS, TLS_SEC_TAG_LIST,
+				sec_tag_list, sizeof(sec_tag_t));
+		if (ret) {
+			LOG_ERR("set tag list failed: %d", -errno);
+			close(data_sock);
+			return -errno;
+		}
+
+	}
+	client.remote.sin_port = htons(data_port);
+	ret = connect(data_sock, (struct sockaddr *)&client.remote,
+		 sizeof(struct sockaddr_in));
+	if (ret < 0) {
+		LOG_ERR("connect(data) failed: %d", -errno);
+		close(data_sock);
+		return -errno;
+	}
+
+	return data_sock;
+}
+
+/**@brief Send FTP message via socket
+ */
+static int do_ftp_send_ctrl(const u8_t *message, int length)
+{
+	int ret;
+	u32_t offset = 0;
+
+	LOG_HEXDUMP_DBG(message, length, "TXC");
+
+	while (offset < length) {
+		ret = send(client.sock, message + offset, length - offset, 0);
+		if (ret < 0) {
+			LOG_ERR("send(ctrl) failed: %d", -errno);
+			ret = -errno;
+			break;
+		}
+		offset += ret;
+		ret = 0;
+	}
+
+	LOG_DBG("CMD sent");
+	return ret;
+}
+
+/**@brief Send FTP data via socket
+ */
+static int do_ftp_send_data(const char *pasv_msg, u8_t *message, u16_t length)
+{
+	int data_sock;
+	int ret;
+	u32_t offset = 0;
+
+	/* Establish data channel */
+	ret = establish_data_channel(pasv_msg);
+	if (ret <= 0) {
+		return ret;
+	}
+
+	LOG_HEXDUMP_DBG(message, length, "TXD");
+
+	data_sock = ret;
+	while (offset < length) {
+		ret = send(data_sock, message + offset, length - offset, 0);
+		if (ret < 0) {
+			LOG_ERR("send(data) failed: %d", -errno);
+			ret = -errno;
+			break;
+		}
+		offset += ret;
+		ret = 0;
+	}
+
+	LOG_DBG("DATA sent");
+	close(data_sock);
+	return ret;
+}
+
+/**@brief Receive FTP message from socket
+ */
+static int do_ftp_recv_ctrl(bool post_result, int success_code)
+{
+	int ret;
+	struct pollfd fds[1];
+
+	memset(ctrl_buf, 0x00, sizeof(ctrl_buf));
+
+	/* Receive FTP control message */
+	fds[0].fd = client.sock;
+	fds[0].events = POLLIN;
+	ret = poll(fds, 1, MSEC_PER_SEC * CONFIG_FTP_LISTEN_TIME);
+	if (ret <= 0) {
+		LOG_ERR("poll(ctrl) failed: (%d)", -errno);
+		return -ETIMEDOUT;
+	}
+	ret = recv(client.sock, ctrl_buf, sizeof(ctrl_buf), 0);
+	if (ret < 0) {
+		LOG_ERR("recv(ctrl) failed: (%d)", -errno);
+		return -errno;
+	}
+	if (ret == 0) {
+		/* Server close connection */
+		return -ECONNRESET;
+	}
+
+	LOG_HEXDUMP_DBG(ctrl_buf, ret, "RXC");
+
+	if (post_result) {
+		client.ctrl_callback(ctrl_buf, ret);
+	}
+
+	LOG_DBG("CTRL received");
+	return parse_return_code(ctrl_buf, success_code);
+}
+
+/**@brief Receive FTP data from socket
+ */
+static void do_ftp_recv_data(const char *pasv_msg)
+{
+	int ret;
+	int data_sock;
+	struct pollfd fds[1];
+	char data_buf[NET_IPV4_MTU];
+
+	/* Establish data channel */
+	ret = establish_data_channel(pasv_msg);
+	if (ret <= 0) {
+		return;
+	}
+
+	/* Receive FTP data message */
+	data_sock = ret;
+	fds[0].fd = data_sock;
+	fds[0].events = POLLIN;
+	do {
+		memset(data_buf, 0x00, sizeof(data_buf));
```

Comment:
Same here, no need to clear the buffer since we overwrite it right after.

Reply:
Fixed